### PR TITLE
Added connect shellcode for thumb/linux.

### PR DESCRIPTION
**Pull request by RobertLarsen**
Tested on an Odroid with both IPv4 and IPv6.

### CODE REVIEW

**Comment by zachriggle** [comment]
It seems like a lot of the code for preparing addrinfo structures will need to be duplicated across a bunch of architectures.  Perhaps we should pull that out and into its own module?

**Comment by RobertLarsen** [comment]
I guess that would be a good idea. I did the same for amd64 and would like it in the i386 as well.
How about I do that as a separete PR?

File: pwnlib/shellcraft/templates/thumb/linux/connect.asm [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        address_family = socket.AF_INET6
+    
+    info = socket.getaddrinfo(host, None, address_family)
+    host = socket.inet_pton(address_family, info[0][4][0])
+    sockaddr  = packing.p16(address_family)
+    sockaddr += packing.p16(socket.htons(port))
+
+    if network == 'ipv4':
+        sockaddr += host
+        sockaddr += '\x00' * (16 - len(sockaddr))
```
Comment (zachriggle) [inline pwnlib/shellcraft/templates/thumb/linux/connect.asm]: sockaddr.ljust(16, '\x00')

Comment (RobertLarsen) [inline pwnlib/shellcraft/templates/thumb/linux/connect.asm]: Nifty. Will remember this when moving the sockaddr code to a common module.

**Pull request merged**